org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/settings/SettingsMergeTest.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.settings;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+
+import org.eclipse.lemminx.extensions.contentmodel.settings.XMLValidationSettings;
+import org.junit.jupiter.api.Test;
+
+public class SettingsMergeTest {
+
+	@Test
+	public void testValidationSettingsMerge() throws IllegalArgumentException, IllegalAccessException {
+		XMLValidationSettings settings = new XMLValidationSettings();
+		mutateValues(settings);
+		XMLValidationSettings mergeTargetSettings = new XMLValidationSettings();
+		mergeTargetSettings.merge(settings);
+		assertEquals(settings, mergeTargetSettings);
+	}
+
+	/**
+	 * Mutates all the values of all the fields, under the assumption that all Object fields are non-null
+	 */
+	public static void mutateValues(Object obj) throws IllegalArgumentException, IllegalAccessException {
+		for (Field field : XMLValidationSettings.class.getFields()) {
```

Comment:
`XMLValidationSettings.class.getFields()` should return all publicly accessible fields, but they're all private, so it's empty for me. I think to do it this way you'd have to use `XMLValidationSettings.class.getDeclaredFields()` in combination with setAccessible(true) which is basically a no-go since Java 9 so I'd probably avoid this.

Alternatively, could you not just use the setters and confirm whether merge does the right thing ?

Reply:
Good catch. I took this route because if we add a new field in the future, then we could end up with the same bug. I'll switch back to using the getters and setters